Sources/BarcodeHeroCore/Extensions/UIImage+Resize.swift

```diff
@@ -12,19 +12,14 @@ import Foundation
 import UIKit
 
 extension UIImage {
-    public func bh_resizedTo(_ imageView: UIImageView?) throws -> UIImage? {
-        guard let imageView = imageView else {
-            throw BHError.imageViewRequired
-        }
-
+    
+    /// Resizes an image based on the size and contentMode of the passed in imageView
+    /// - Parameter imageView: The imageView whose size and contentMode should be referenced
+    public func bh_resizedTo(_ imageView: UIImageView) throws -> UIImage? {
```

Comment:
I think there are two mentalities we can have with any SDK interface:

- Provide optionality and catch `nil` values for the consumer, to avoid any consumers being bloated with statements that will be doing the same thing within this block.
- Reduce optionality and require non-nil values, which puts the onus on the consumer to handle any unwrapping.

I typically lean towards the former (hence why these are optional) when working with UIKit objects especially. Can you tell me how you view it from your perspective?

Reply:
I lean on the latter since it seems strange to me to see a method signature that says an argument is optional and then immediately errors on me saying I provided an argument that, while it fits the method signature, is incorrect for the method to begin doing anything useful.

Reply:
That's a great explanation. Optional might make sense if it can actually work with it somehow, but in providing an error, it's punishing the user for leaning into it.

Would you rather `return nil` here and not throw an error, or would you rather drop it entirely? If we `return nil`, how does that change your explanation?

Reply:
If this changed from throwing when a null argument was passed in to returning nil in that case I'd still feel the same way. I still feel if a non-optional argument is required to at least attempt performing some logic, then force the user to pass in a non-optional

Reply:
Sounds good! I'm still on the fence with considering this a hard-and-fast rule everywhere, but in this case (being an extension especially) I'm inclined to agree.